node/src/bin/manager.rs

```diff
@@ -514,21 +541,21 @@ impl Context {
         self.node_id.clone()
     }
 
-    fn primary_pool(self) -> ConnectionPool {
+    fn primary_pool(&self) -> ConnectionPool {
         let primary = self.config.primary_store();
         let pool = StoreBuilder::main_pool(
             &self.logger,
             &self.node_id,
             PRIMARY_SHARD.as_str(),
             primary,
-            self.registry,
+            self.metrics_registry(),
             Arc::new(vec![]),
         );
         pool.skip_setup();
         pool
     }
 
-    fn subgraph_store(self) -> Arc<SubgraphStore> {
+    fn subgraph_store(&self) -> Arc<SubgraphStore> {
```

Comment:
The reason all these methods take `self` and not `&self` is to make sure that `StoreBuilder` doesn't get called more than once since that would create more pools than expected. You could e.g. call `ethereum_networks()` first and `store()` second in your code below, and only `await` on the eth networks in the `if let Some(..)`. Generally, it would be nicer if we only connected to the network the user needs instead of all of them, but that can wait for a future enhancement.

Reply:
Got it!
That makes a lot of sense now.
I'll revert that change and add a comment explaining the rationale for moving `self`.

Reply:
Fixed in 569e9bbd7